applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -119,16 +119,19 @@ void Define2DWakeProcess::SetWakeDirectionAndNormal()
 // This function finds and saves the trailing edge for further computations
 void Define2DWakeProcess::SaveTrailingEdgeNode()
 {
+    KRATOS_ERROR_IF(mrBodyModelPart.NumberOfNodes() == 0) << "There are no nodes in the body_model_part!"<< std::endl;
+
     double max_x_coordinate = std::numeric_limits<double>::lowest();
-    NodeType* trailing_edge_node;
+    NodeType& trailing_edge_node = mrBodyModelPart.GetNode(1);
```

Comment:
I think you can save the reference to this auxiliary node by just saving the `Id`. Then you can do `mpTrailingEdgeNode = &mrBodyModelPart.GetNode(aux_id);` and use the member pointer to do the `mpTrailingEdgeNode->SetValue(TRAILING_EDGE,true); `. I think that this makes also possible to have a `const` iterator in the loop.